modules/_labs/select/react/lib/SelectBase.tsx

```diff
@@ -363,105 +326,133 @@ export default class SelectBase extends React.Component<SelectBaseProps> {
     });
   };
 
-  renderOption: RenderOptionFunction = option => {
+  const defaultRenderOption: RenderOptionFunction = option => {
     return <div>{option.label}</div>;
   };
 
-  public render() {
-    const {
-      'aria-labelledby': ariaLabelledBy,
-      'aria-required': ariaRequired,
-      buttonRef,
-      disabled,
-      error,
-      focusedOptionIndex,
-      grow,
-      inputRef,
-      isEmpty,
-      isMenuFlipped,
-      isMenuHidden,
-      isMenuHiding,
-      onChange,
-      onKeyDown,
-      onMenuBlur,
-      onMenuCloseOnEscape,
-      options,
-      required,
-      shouldMenuAnimate,
-      shouldMenuAutoFlip,
-      shouldMenuAutoFocus,
-      value,
-
-      // Strip unneeded props on button from elemProps
-      onOptionSelection,
-
-      ...elemProps
-    } = this.props;
-
-    // Use default renderOption if renderOption prop isn't provided
-    const renderOption = this.props.renderOption || this.renderOption;
-
-    // Do a bit of error-checking in case options weren't provided
-    const selectedOption = !isEmpty ? options[getCorrectedIndexByValue(options, value)] : null;
-    const selectedOptionLabel = selectedOption ? selectedOption.label : '';
-    const selectedOptionValue = selectedOption ? selectedOption.value : '';
-
-    return (
-      <SelectWrapper grow={grow} disabled={disabled}>
-        <SelectButton
-          aria-expanded={!isMenuHidden ? 'true' : undefined}
-          aria-haspopup="listbox"
-          aria-controls={!isMenuHidden ? this.menuId : undefined}
-          disabled={disabled}
+  // If the focused option changed, scroll the newly focused option into view (if
+  // necessary) but do NOT center it
+  useLayoutEffect(() => {
+    const focusedOption = focusedOptionRef.current;
+
+    if (focusedOption) {
+      // TODO: Figure out why rAF is necessary here (and below) in order for the
+      // Select States Menu On story to render correctly in IE. Without rAF,
+      // the DOM measurements performed in scrollIntoViewIfNeeded are incorrect
+      // and we perform unnecessary scrolling BUT ONLY IN THIS STORY and ONLY IN
+      // IE. Everywhere else (in IE and in other browsers), everything works
+      // fine without rAF.
+      const animateId = requestAnimationFrame(() => {
+        // We cannot use the native Element.scrollIntoView() here because it
+        // doesn't work properly with the portalled menu: when using the keyboard
+        // to advance focus through the options, using scrollIntoView to keep the
+        // newly focused option in view also scrolls the ENTIRE page. Instead, we
+        // call our own scrollIntoViewIfNeeded function.
+        scrollIntoViewIfNeeded(focusedOption, false);
+      });
```

Comment:
This scrolling code works fine except for the `Select States Menu On` visual testing story in IE. Without `requestAnimationFrame`, the menu is scrolled slightly further down than it should be, which creates a visual regression. The scrolling code is wrapped in `useLayoutEffect` which should ensure proper measurements, but that doesn't appear to be happening in IE. I'm still investigating.

Reply:
Added a bit of extra info in the code comments for this rAF call. Resolving this and adding an updated comment.